test/e2e/storage/testsuites/provisioning.go

```diff
@@ -198,6 +208,15 @@ func testProvisioning(input *provisioningTestInput) {
 		input.pvc.Spec.VolumeMode = &block
 		TestDynamicProvisioning(input.testCase, input.cs, input.pvc, input.sc)
 	})
+
+	It("should provision storage with snapshot data source [Feature:VolumeSnapshotDataSource]", func() {
+		if !input.dInfo.Capabilities[CapDataSource] {
+			framework.Skipf("Driver %q does not support populate data from snapshot - skipping", input.dInfo.Name)
+		}
+
+		input.pvc.Spec.DataSource = prepareDataSourceForProvisioning(input.testCase, input.cs, input.dc, input.pvc, input.sc, input.vsc)
+		TestDynamicProvisioning(input.testCase, input.cs, input.pvc, input.sc)
+	})
```

Comment:
Please don't extend `TestDynamicProvisioning` with test-case specific code. Instead set `input.testCase.CheckPV` and implement your custom check there. Does your test need the write/read check? Probably not. You can make it run faster by setting `input.testCase.SkipWriteReadCheck = true`.

@msau42 pointed out that `TestDynamicProvisioning` is getting overloaded with different checks in https://github.com/kubernetes/kubernetes/pull/72002#discussion_r248153684 and now I am trying to clean that up in that PR.